model-optimizer/extensions/back/names_uniqueness_check.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from extensions.back.pass_separator import BackFinish
+from mo.back.replacement import BackReplacementPattern
+from mo.graph.graph import Graph, rename_node
+
+
+class NamesUniquenessCheck(BackReplacementPattern):
+    enabled = True
+
+    def run_after(self):
+        return [BackFinish]
+
+    def run_before(self):
+        return []
+
+    def find_and_replace_pattern(self, graph: Graph):
+        names_dict = {}
+        not_unique_names = []
+        for node in graph.get_op_nodes():
+            node_name = node.soft_get('name', node.id)
+            if node_name not in names_dict.keys():
+                names_dict[node_name] = [node]
+            else:
+                names_dict[node_name].append(node)
+                not_unique_names.append(node_name)
+
+        if not_unique_names:
+            not_unique_names = set(not_unique_names)
+            for name in not_unique_names:
+                for idx, node in enumerate(names_dict[name]):
+                    rename_node(node, name + '_{}'.format(idx))
```

Comment:
We rename all nodes with non unique name. Is it will be a problem if one of such nodes is output node for which we save framework name when possible?

Reply:
Also, we need to check that the new name which you try to assign to a node `name + '_{}'.format(idx)` is actually unique. There is no such check now.

Reply:
Done. Added a test for this case

Reply:
Should we check that we have not several Results with the same name? I believe it should be always true, so it is just minor note.